File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.scala
/**
 * Definition for a binary tree node.
 * class TreeNode(_value: Int = 0, _left: TreeNode = null, _right: TreeNode = null) {
 *   var value: Int = _value
 *   var left: TreeNode = _left
 *   var right: TreeNode = _right
 * }
 */

import scala.collection.mutable._

object Solution {
    def levelOrder(root: TreeNode): List[List[Int]] = {
        var res: List[List[Int]] = List()
        val queue: Queue[TreeNode] = Queue()
        
        if (root != null) { queue.enqueue(root) }
        
        while(queue.nonEmpty) {
            val qs = queue.length
            var list: List[Int] = List()
            
            for(i <- 1 to qs) {
                val n = queue.dequeue
                list = list :+ n.value
                if (n.left != null) { queue.enqueue(n.left) }
                if (n.right != null) { queue.enqueue(n.right) }
            }
            
            res = res :+ list
        }
        
        res
    }
}

File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        Queue<TreeNode> queue = new LinkedList<>();
        if (root != null) queue.offer(root);
        
        while(!queue.isEmpty()) {
            int sz = queue.size();
            List<Integer> lvl = new ArrayList<>();
            
            for(int i = 0; i < sz; i++) {
                TreeNode node = queue.poll();
                lvl.add(node.val);
                if (node.left != null) queue.offer(node.left);
                if (node.right != null) queue.offer(node.right);
            }
            
            res.add(lvl);
        }
        
        return res;
    }
}